Add consul API retries

gdavison opened the pull request:
When starting up a new machine and creating an ACL, the Consul server is not always ready to serve requests by the time the ACL is ready to be created. This causes intermittent failures in integration tests, and likely in real-world use.

Two retries with an interval of 0.5 seconds seems to work.

Also addressed some `cookstyle` issues

legal90 commented on libraries/consul_acl.rb:
```diff
@@ -90,10 +90,35 @@ def configure_diplomat
+          raise if attempts > opts[:attempts]
+
+          # Sleep before the next retry if the option was given
+          sleep 0.5
```
Should it be `sleep opts[:sleep]` instead?

gdavison replied:
Yes :)

legal90 commented on libraries/consul_acl.rb:
```diff
@@ -90,10 +90,35 @@ def configure_diplomat
+      def retry_block(opts = {}, &_block)
+        opts = {
+          sleep: 0, # Seconds to sleep between attempts
+        }.merge(opts)
```
I think, we should also define a define for `:attempts` option here. Otherwise, if it's not passed to the method, `opts[:attempts]` will be `nil` and line 115 will fail:
```
ArgumentError: comparison of Fixnum with nil failed
```

gdavison replied:
Yes

legal90 commented on libraries/consul_acl.rb:
```diff
@@ -90,10 +90,35 @@ def configure_diplomat
+
+        begin
+          return yield attempts
+        rescue Diplomat::UnknownStatus
```
Should it be `rescue *opts[:catch]` instead?

gdavison replied:
No, this will be the only error we want to retry on

legal90 replied:
Yeah, now it's fine

gdavison commented:
@legal90 I've removed some bits I should have removed when I copied the code. It was originally much more general, but I've made it specific here.

I haven't tried it for real, because we don't use ACLs yet. My main reason for adding this was to remove the intermittent test failures. I've added an additional attempt, so it will give 1.5 seconds for the consul service to start. It can definitely be tuned if needed.

legal90 commented on libraries/consul_acl.rb:
```diff
@@ -90,10 +90,35 @@ def configure_diplomat
+          sleep:    0, # Seconds to sleep between attempts
+        }.merge(opts)
+
+        attempts = 1
```
I would suggest renaming `attempts` to something like `count`, or `opt[:attempts]` to `opt[:max_attempts]` in order to avoid the confusion between these two variables.

gdavison commented:
Hi @johnbellone, any feedback on the PR?

gdavison commented:
Fixes #461

gdavison commented:
@johnbellone, yes, it would (still) fail in this case if the local Consul agent were not yet available. Before this change, there would be a race condition which caused intermittent failures, at least is the tests. Now, the behaviour is to try three times before failing.
